### PR TITLE
Added deploying infrastructure from release branches

### PR DESCRIPTION
**Pull request by eponomar**
- Changed pollers strategy - GitPoller works only for release branches,
  GitHubPullRequestPoller works for all pull requests
- Added "target_branch" properties to correct extracting infrastructure\product
  repositories
- Added dynamic factories

### CODE REVIEW

File: build_scripts/build_runner.py [hunk 1]
```diff
@@ -637,7 +638,9 @@ def _extract(self):
                     if self.repo_url:
                         data['url'] = self.repo_url
                 elif not data.get('branch'):
-                    if MediaSdkDirectories.is_release_branch(branch):
+                    if self.target_branch and MediaSdkDirectories.is_release_branch(self.target_branch):
+                        data['branch'] = self.target_branch
+                    elif MediaSdkDirectories.is_release_branch(branch) and not self.target_branch:
```
Comment (dlobanox) [inline build_scripts/build_runner.py]: "and not self.target_branch" is not needed.

File: build_scripts/build_runner.py [hunk 2]
```diff
@@ -1040,6 +1043,8 @@ def main():
                         default=[target_arch.value for target_arch in TargetArch],
                         choices=[target_arch.value for target_arch in TargetArch],
                         help='Architecture of target platform')
+    parser.add_argument('-tb', "--target-branch", default=None,
```
Comment (dlobanox) [inline build_scripts/build_runner.py]: remove "default=None"

File: bb/master/master.cfg [hunk 3]
```diff
@@ -232,45 +274,62 @@ c["services"] = [
 c["change_source"] = []
 
 
-def pull_request_filter(pull_request_message):
-    # Include only changes from fork repositories
-    if pull_request_message["head"]["repo"] and pull_request_message["head"]["repo"][
-        "full_name"] == config.GITHUB_REPOSITORY:
+def is_msdk_org_member(pull_request_message):
+    commit_owner = pull_request_message['head']['user']['login']
+    github_member_url = f"https://api.github.com/orgs/Intel-Media-SDK/members/{commit_owner}?access_token={config.GITHUB_TOKEN}"
```
Comment (dlobanox) [inline bb/master/master.cfg]: Hardcode. Please, move to a common part or add TODO.

File: bb/master/master.cfg [hunk 4]
```diff
@@ -232,45 +274,62 @@ c["services"] = [
 c["change_source"] = []
 
 
-def pull_request_filter(pull_request_message):
-    # Include only changes from fork repositories
-    if pull_request_message["head"]["repo"] and pull_request_message["head"]["repo"][
-        "full_name"] == config.GITHUB_REPOSITORY:
+def is_msdk_org_member(pull_request_message):
```
Comment (azhogov) [inline bb/master/master.cfg]: it seems duplicate

File: build_scripts/build_runner.py (hunk 2 above)
Comment (eponomar) [reply build_scripts/build_runner.py]: Done

File: bb/master/master.cfg (hunk 3 above)
Comment (eponomar) [reply bb/master/master.cfg]: Done

File: bb/master/master.cfg [hunk 5]
```diff
@@ -40,30 +44,37 @@ class Stage(Enum):
     PACK = "pack"
     COPY = "copy"
 
+def get_repository_name_by_url(repo_url):
+    # Some magic for getting repository name from url, by example
```
[the remark this answers is not in the document]
Comment (eponomar) [reply bb/master/master.cfg]: Comment updated

File: bb/master/master.cfg [hunk 6]
```diff
@@ -40,30 +44,37 @@ class Stage(Enum):
...
 def get_infrastructure_deploying_cmd(props):
     infrastructure_deploying_cmd = [
         config.RUN_COMMAND, 'extract_repo.py',
         '--repo-name', 'OPEN_SOURCE_INFRA',
-        '--root-dir', props.getProperty("builddir"),
-        '--branch', props.getProperty("branch")
-    ]
+        '--root-dir', props.getProperty("builddir")]
 
     # Changes in product configs from Intel-Media-SDK/product-configs only will be deployed
```
[the remark this answers is not in the document]
Comment (eponomar) [reply bb/master/master.cfg]: Comment updated

File: bb/master/master.cfg [hunk 7]
```diff
@@ -79,6 +90,38 @@ def get_changed_repo(props):
     return f"{repo_name}:{branch}:{revision}"
 
 
+class StepsGenerator(steps.BuildStep):
+    """
+    Add static steps dynamically in runtime
```
[the remark this answers is not in the document]
Comment (eponomar) [reply bb/master/master.cfg]: Comment updated

**Pull request merged**